Octree/include/CGAL/Octree.h

```diff
@@ -0,0 +1,646 @@
+  typedef typename Kernel::Iso_cuboid_3 Iso_cuboid;
+  typedef typename Kernel::Sphere_3 Sphere;
+  typedef typename CGAL::Bbox_3 Bbox;
+  typedef typename PointRange::iterator Range_iterator;
+  typedef typename std::iterator_traits<Range_iterator>::value_type Range_type;
+
+private: // data members :
+
+  Node m_root;                      /* root node of the octree */
+  uint8_t m_max_depth_reached = 0;  /* octree actual highest depth reached */
+
+  PointRange &m_ranges;              /* input point range */
+  PointMap m_points_map;          /* property map: `value_type of InputIterator` -> `Point` (Position) */
+
+  Point m_bbox_min;                  /* input bounding box min value */
+  FT m_bbox_side;              /* input bounding box side length (cube) */
+
+  std::vector<FT> m_side_per_depth;      /* side length per node's depth */
+
+public:
+
+  /// \name Construction, Destruction
+  /// @{
+
+  /*!
+   * \brief Create an octree from a collection of points
+   *
+   * The resulting octree will have a root node with no children that contains the points passed.
+   * That root node will have a bounding box that encloses all of the points passed,
+   * with padding according to the enlarge_ratio
+   * This single-node octree is valid and compatible with all Octree functionality,
+   * but any performance benefits are unlikely to be realized unless the tree is refined.
+   *
+   * \param point_range random access iterator over the indices of the points
+   * \param point_map maps the point indices to their coordinate locations
+   * \param enlarge_ratio the degree to which the bounding box should be enlarged
+   */
+  Octree(
+          PointRange &point_range,
+          PointMap &point_map,
+          const FT enlarge_ratio = 1.2) :
+          m_ranges(point_range),
+          m_points_map(point_map) {
+
+    // compute bounding box that encloses all points
+    Iso_cuboid bbox = CGAL::bounding_box(boost::make_transform_iterator
+                                                 (m_ranges.begin(),
+                                                  CGAL::Property_map_to_unary_function<PointMap>(
+                                                          m_points_map)),
+                                         boost::make_transform_iterator
+                                                 (m_ranges.end(),
+                                                  CGAL::Property_map_to_unary_function<PointMap>(
+                                                          m_points_map)));
+
+    // Find the center point of the box
+    Point bbox_centroid = midpoint(bbox.min(), bbox.max());
+
+    // scale bounding box to add padding
+    bbox = bbox.transform(Aff_transformation_3<Kernel>(SCALING, enlarge_ratio));
+
+    // Convert the bounding box into a cube
+    FT x_len = bbox.xmax() - bbox.xmin();
+    FT y_len = bbox.ymax() - bbox.ymin();
+    FT z_len = bbox.zmax() - bbox.zmin();
+    FT max_len = std::max({x_len, y_len, z_len});
```

Comment:
`std::max` -> `(std::max)`   to avoid problems with VC++ .   Is the same for `min`

Reply:
This should be fixed now, I think I only use `std::max` in one place. I'm surprised windows.h would contain such an obvious naming conflict!